### PR TITLE
Big changes to documentation generator

### PR DESCRIPTION
**Pull request by compleatang**

### CODE REVIEW

File: go/docs/generator.go_ [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	tmpl, err := docs.GenerateCommandsTemplate()
+	if err != nil {
+		panic(err)
+	}
+
+	// render templates
+	// if err = RenderFiles(pm, tmpl); err != nil {
+		panic(err)
+	}
+}
```
Comment (pietv) [inline go/docs/generator.go_]: This won't compile

File: go/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+These are as follows:
+
+* `title`: Title case a string (wrapper for strings.Title)
+* `replace`: Replace a string (wrapper for strings.Replace)
+* `chomp`: Trim trailing whitespace (wrapper for strings.TrimSpace)
+* `handle_file`: Given a string, render its `.md` filename
+* `handle_link`: Given a filename, render it's link
+* `insert_definition`: Given a file which is within the `./definitions` directory and the name of a struct, this will output the entire struct and add the appropriate code fences (triple backticks + language modifier) (note, both variables must be quoted).
+* `insert_bash_lines`: Given a file which is within the `./docs/tests` directory and a set of line numbers in the form `"10-16"` or simply `"16"` this function will embed the appropriate lines with the appropriate code fences (again, quote everything here).
+* `insert_file`: Given a file name which is within the `./docs/tests` directory, renders the entire file with the appropriat code fences and also download (`curl`) command.
```
Comment (pietv) [inline go/docs/README.md]: *appropriate

File: go/docs/docs_generator.go [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+		"title":       strings.Title,
+		"replace":     strings.Replace,
+		"chomp":       strings.TrimSpace,
+		"handle_file": handle_file,
+		"handle_link": handle_link,
+		"insert_definition": insert_definition,
+		"insert_bash_lines": insert_bash_lines,
+		"insert_file": insert_file,
+	}
+
```
Comment (pietv) [inline go/docs/docs_generator.go]: `gofmt`

File: go/docs/generator.go_ [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+
+	// commands "github.com/eris-ltd/eris-pm/cmd"
+	// "github.com/eris-ltd/eris-pm/version"
+	"github.com/eris-ltd/common/go/docs"
```
Comment (pietv) [inline go/docs/generator.go_]: garbage

File: go/docs/generator.go_ [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+	return nil
+}
+
+func main() {
+	// repos should populate the top level command object
+	// pm := commands.EPMCmd
+	// commands.InitEPM()
+	// commands.AddGlobalFlags()
+
```
Comment (pietv) [inline go/docs/generator.go_]: garbage

**Comment by pietv** [comment]
LGTM!

File: go/docs/generator.go_ (hunk 1 above)
Comment (compleatang) [inline go/docs/generator.go_]: ya. exactly!!!! that way folks will remember to update ;)

Comment (pietv) [inline go/docs/generator.go_]: Cool :)

File: go/docs/generator.go_ [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/spf13/cobra"
+)
+
+// repos should customize the next two lines
+var Description = "DESCRIPTION!!!" // should match the docs site name
+var RenderDir = fmt.Sprintf("./docs/documentation/REPO/%s/", version.VERSION) // should be the "shortversion..."
+
+// The below variables should be updated only if necessary
+var Specs = []*docs.Entry{}
```
Comment (pietv) [inline go/docs/generator.go_]: Inconsistent comment style. 

The accepted style is to begin the comment with a capital letter and end with a period: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Also this: https://blog.golang.org/godoc-documenting-go-code

But that's a minor thing.

File: go/docs/docs_generator.go [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+* [{{ title .Title }}]({{ .URL }})
+{{- end }}
+{{ end }}
+`
+
+	return template.New("entryGenerator").Funcs(funcMap).Parse(templateText)
+}
+
+func RenderEntry(this_entry *Entry) error {
+	out_file, err := os.Create(this_entry.FileName)
```
Comment (pietv) [inline go/docs/docs_generator.go]: snake case ;) https://golang.org/doc/effective_go.html#mixed-caps

**Pull request merged**